cargo-shuttle/src/args.rs

```diff
@@ -215,90 +217,80 @@ pub struct RunArgs {
-    pub thruster: bool,
-    /// Initialize without a framework
-    #[arg(long, conflicts_with_all = &["actix_web","axum", "rocket", "tide", "tower", "poem", "warp", "salvo", "serenity", "poise", "thruster"])]
-    pub no_framework: bool,
+    /// Initialize the project with a template
+    #[arg(long, short, value_enum)]
+    pub template: Option<InitTemplateArg>,
     /// Whether to create the environment for this project on shuttle
     #[arg(long)]
-    pub new: bool,
+    pub create_env: bool,
     #[command(flatten)]
     pub login_args: LoginArgs,
     /// Path to initialize a new shuttle project
-    #[arg(default_value = ".", value_parser = OsStringValueParser::new().try_map(parse_init_path) )]
+    #[arg(default_value = ".", value_parser = OsStringValueParser::new().try_map(parse_init_path))]
     pub path: PathBuf,
 }
 
+#[derive(ValueEnum, Clone, Debug)]
+pub enum InitTemplateArg {
+    /// Initialize with actix-web framework
+    ActixWeb,
+    /// Initialize with axum framework
+    Axum,
+    /// Initialize with poem framework
+    Poem,
+    /// Initialize with poise framework
+    Poise,
+    /// Initialize with rocket framework
+    Rocket,
+    /// Initialize with salvo framework
+    Salvo,
+    /// Initialize with serenity framework
+    Serenity,
+    /// Initialize with tide framework
+    Tide,
+    /// Initialize with thruster framework
+    Thruster,
+    /// Initialize with tower framework
+    Tower,
+    /// Initialize with warp framework
+    Warp,
+    /// Initialize with no template
+    None,
+}
+
 impl InitArgs {
-    pub fn framework(&self) -> Option<Framework> {
-        if self.actix_web {
-            Some(Framework::ActixWeb)
-        } else if self.axum {
-            Some(Framework::Axum)
-        } else if self.rocket {
-            Some(Framework::Rocket)
-        } else if self.tide {
-            Some(Framework::Tide)
-        } else if self.tower {
-            Some(Framework::Tower)
-        } else if self.poem {
-            Some(Framework::Poem)
-        } else if self.salvo {
-            Some(Framework::Salvo)
-        } else if self.poise {
-            Some(Framework::Poise)
-        } else if self.serenity {
-            Some(Framework::Serenity)
-        } else if self.warp {
-            Some(Framework::Warp)
-        } else if self.thruster {
-            Some(Framework::Thruster)
-        } else if self.no_framework {
-            Some(Framework::None)
-        } else {
-            None
-        }
+    /// `None` -> No template chosen, ask for it
+    ///
+    /// `Some(Template::None)` -> Init with a blank cargo project
+    pub fn framework(&self) -> Option<Template> {
+        // Why separate enums?
+        // Future might have more options that pre-defined templates
+        self.template.as_ref().map(|t| {
+            use InitTemplateArg::*;
```

Comment:
Are you referring to the wildcard enum import? That's just my personal preference in a situation like this, I can remove it if you want. (this will probably be reworked in the upcoming changes anyways...)